addin-proximity/app/scripts/main.js

```diff
@@ -313,23 +320,48 @@ geotab.addin.proximity = () => {
                 }
             })
 
+            const getBatchSize = (deviceLength, utcFrom, utcTo) => {
+                //variable created to limit the api multiCall size. 
+                //1209600000 is 2 weeks in milliseconds 
+                // 1000 is device limit for proximity add-in
+                let maxBatchSize = 400;
+                let twoWeeksInSeconds = 1209600000;
+                let deviceLimit = 1000;
+                let variable = twoWeeksInSeconds * maxBatchSize * deviceLimit;
+
+                let batchSize = Math.round(variable / ((new Date(utcTo) - new Date(utcFrom)) * deviceLength));
+                if (batchSize > maxBatchSize) {
+                    batchSize = maxBatchSize;
+                }
+                return batchSize;
+            }
+
             let getDeviceLogs = device => new Promise(async (resolve) => {
                 var request = [];  
                 var temp = [];
                 var totalPoints = 0;
-
-                for(var j=0;j<device.length;j++){
-                    temp.push(buildGetRequest(device[j].id, utcFrom, utcTo));
-                    if(temp.length == 400){
-                        request.push(temp);
-                        temp = [];
+                let minBatchSize = 50;
+
+                let batchSize = getBatchSize(device.length, utcFrom, utcTo);
+                console.log(batchSize);
+                if (batchSize >= minBatchSize) {
+                    for(var j=0;j<device.length;j++){
```

Comment:
inconsistent style (spaces).
mixing of `let` and `var`, idealy should use `let` in all new code and we update old code at some point.

Reply:
Will fix that in the next commit.